ZeeHomeScreen/Classes/Factory/ComponenttFactory.m

```diff
@@ -22,52 +22,6 @@ @implementation ComponenttFactory
 
 #pragma mark - Definitions
 
-+ (UIViewController <ComponentProtocol>  *)componentViewControllerWithComponentModel:(ComponentModel *)componentModel
```

Comment:
> In class-based programming, the factory method pattern is a creational pattern that uses factory methods to deal with the problem of creating objects without having to specify the exact class of the object that will be created.

Factory should never, ever, alter different object passed do the method as arguments. According to single responsibility principle, the sole purpose of factory should be creating new objects, not adding them to a view hierarchy, which in this case will also cause a memory leak, because factory already adds a reference of created object to parent view controller and never takes care of releasing it.

As a solution I just got rid of this method, because in my opinion it is broken by design.

Reply:
SRP fix.